src/main/java/com/gooddata/executeafm/afm/OverPeriodMeasureDefinition.java

```diff
@@ -0,0 +1,166 @@
+public class OverPeriodMeasureDefinition implements MeasureDefinition, Serializable {
+
+    private static final long serialVersionUID = -8904516814279504098L;
+
+    static final String NAME = "overPeriodMeasure";
+
+    private final String measureIdentifier;
+    private final List<OverPeriodDateAttribute> dateAttributes;
+
+    /**
+     * Create a new instance of {@link OverPeriodMeasureDefinition}.
+     *
+     * @param measureIdentifier
+     *         The local identifier of the measure this PoP measure refers to. The parameter must not be null.
+     * @param dateAttributes
+     *         The date attributes that defines how this measure will be shifted in time. The parameter must not be null.
+     *
+     * @throws IllegalArgumentException
+     *         Thrown when {@code dateAttributes} list is empty or required parameter is null.
+     */
+    @JsonCreator
+    public OverPeriodMeasureDefinition(
+            @JsonProperty("measureIdentifier") final String measureIdentifier,
+            @JsonProperty("dateAttributes") final List<OverPeriodDateAttribute> dateAttributes) {
+        this.measureIdentifier = notNull(measureIdentifier, "measureIdentifier");
+        this.dateAttributes = notEmpty(dateAttributes, "dateAttributes");
+    }
+
+    /**
+     * The method is not supported by the object.
+     * Use {@link #getObjQualifiers()} instead.
+     *
+     * @return throws {@link UnsupportedOperationException}
+     *
+     * @throws UnsupportedOperationException
+     *         The exception is thrown every time the method is called.
+     * @deprecated Use {@link #getObjQualifiers()} instead.
+     */
+    @Override
+    @Deprecated
+    public ObjQualifier getObjQualifier() {
+        throw new UnsupportedOperationException("The method is not supported by the object!");
+    }
+
+    @Override
+    public Collection<ObjQualifier> getObjQualifiers() {
+        return this.dateAttributes.stream()
+                .map(OverPeriodDateAttribute::getAttribute)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * The method is not supported by the object.
+     * Use {@link #withObjUriQualifiers(ObjQualifierConverter)} instead.
+     *
+     * @param uriQualifier
+     *         The URI object qualifier.
+     *
+     * @return throws {@link UnsupportedOperationException}
+     *
+     * @throws UnsupportedOperationException
+     *         The exception is thrown every time the method is called.
+     * @deprecated Use {@link #withObjUriQualifiers(ObjQualifierConverter)} instead.
+     */
+    @Override
+    @Deprecated
+    public MeasureDefinition withObjUriQualifier(final UriObjQualifier uriQualifier) {
+        throw new UnsupportedOperationException("The method is not supported by the object!");
+    }
+
+    @Override
+    public MeasureDefinition withObjUriQualifiers(final ObjQualifierConverter objQualifierConverter) {
+        notNull(objQualifierConverter, "'objQualifierConverter' must not be null!");
```

Comment:
no need to write all the message when using this validator

Reply:
I found two placed where I forgot to update the message. Both of them are fixed.